includes/class-indieauth-client-discovery.php

```diff
@@ -74,45 +75,63 @@ private function parse( $url ) {
 			return $response;
 		}
 
-		$content = wp_remote_retrieve_body( $response );
-
-		if ( class_exists( 'Masterminds\\HTML5' ) ) {
-			$domdocument = new \Masterminds\HTML5( array( 'disable_html_ns' => true ) );
-			$domdocument = $domdocument->loadHTML( $content );
-		} else {
-			$domdocument = new DOMDocument();
-			libxml_use_internal_errors( true );
-			if ( function_exists( 'mb_convert_encoding' ) ) {
-				$content = mb_convert_encoding( $content, 'HTML-ENTITIES', mb_detect_encoding( $content ) );
+		$content_type = wp_remote_retrieve_header( $response, 'content-type' );
+		if ( 'application/json' === $content_type ) {
+			$json = json_decode( wp_remote_retrieve_body( $response ), true );
```

Comment:
```suggestion
			/** @var $json array{
			 *  client_id: string,
			 *     client_name: string,
			 *     logo_uri: string
			 * }
			 */
			$json = json_decode( wp_remote_retrieve_body( $response ), true );
```

Reply:
I like using comments to describe what we expect to be in json like this since it makes identifying and using them slightly easier.